docs/userguide/datafiles.rst

```diff
@@ -337,6 +351,40 @@ Any files that match these patterns will be *excluded* from installation,
 even if they were listed in ``package_data`` or were included as a result of using
 ``include_package_data``.
 
+.. _interplay_package_data_keywords:
+
+Interplay between these keywords
+--------------------------------
+
+Meanwhile, to further clarify the interplay between these three keywords,
+to include certain data file into the source distribution, the following
+logic condition has to be met::
+
+    m or (p and not e)
+
+In plain language, the file should be either: 1. included in ``MANIFEST.in``;
+or 2. selected by ``package-data`` AND not excluded by ``exclude-package-data``.
+
+To include some data file into the ``.whl``::
+
+    (not e) and ((i and m) or p)
+
+In plain language, the file should not be excluded by ``exclude-package-data``
+(highest priority), and should be either: 1. selected by ``package-data``; or
+2. selected by ``MANIFEST.in`` AND use ``include-package-data = true``.
+
+**Notation**::
+
+    i - "include-package-data = true" is set
+    e - file selected by "exclude-package-data"
+    p - file selected by "package-data"
+    m - file included in "MANIFEST.in"
+
+.. note::
+    Different versions of ``setuptools`` might behave differently. The above
+    description applies to versions after ``58.5.3`` (exclusive). For information
+    on the behavior of earlier versions and more details, please refer to the
+    `GitHub repository <https://github.com/abravalheri/experiment-setuptools-package-data>`_.
```

Comment:
Not sure if we need this note, because that was a bug in prior versions (so we don't really need to document bug behaviour).

We could instead use something like:

```suggestion
   Please make sure to use a recent version of setuptools, to avoid potential problems and bugs.
```

Or even omit it completely.

Reply:
Thanks for the input, let's remove it then. I was not aware of the background and thought we have to maintain backwards awareness (didn't know it was a buggy behaviour).